### ISSUE TITLE
Button Click with Mouse initiates a onclick, but the State (both Visual and Value) does not change

### ISSUE THREAD

**Issue by danielHalan** [ghinda/css-toggle-switch#44]
This happens when the user clicks and "drags"  the mouse as-if-selecting the text, (and very little movement is needed so not always any selection is created), the Button does not get selected/checked, but the "onclick" is triggered. (tested in Chrome)

We have an OnClick on both labels, with different input param which is used to change the state, 
i.e.
<label for="typeAdditive" onclick="setupType('Additive')">Additive</label> 

A work-around would have been to check the "checked" state in the onClick, but unfortunately that one have not yet changed at the moment of the click-event.

**Expected States**
![image](https://cloud.githubusercontent.com/assets/641348/9992997/a199c978-609f-11e5-9e8f-f4a520387daa.png)

![image](https://cloud.githubusercontent.com/assets/641348/9993007/ac45b58a-609f-11e5-9d19-182106c91283.png)

**Error State** 
![image](https://cloud.githubusercontent.com/assets/641348/9993032/c1d48f16-609f-11e5-8b74-9a819cd6be21.png)

**Comment by ghinda** [comment]
The problem is that if you select the label text and then release the mouse button, the radio button is not getting selected.

I think it's caused by the actual text selection, because if I disable it with css, it seems to work:
http://jsbin.com/duwono/edit?html,css,output

Could you try out the demo above and let me know if it works better for you?

I would like to avoid adding a JavaScript dependency, and disabling text selection could be an accessibility issue, so I would still like to find a better way.

**Comment by danielHalan** [comment]
Yes, the css trick is keeping the state right... works good as an initial workaround at least, but yes it can be an issue with accessibility... 

One road to explore is onclick/checkstate life-cycle, try to fire event after the state has changed, or an event with the new state as param

**Comment by ghinda** [comment]
I'm going to close this for now, since I couldn't find a different CSS work-around, and don't want to introduce accessibility issues because of disabling text selection.

If more people report this, I'll probably add `user-select: none` by default.

In the meanwhile, you can use:

```
.switch-light,
.switch-toggle label {
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}
```

**Comment by ghinda** [comment]
I'm working on a fix for this on the `material` branch, so I'm re-opening it.

**Comment by danielHalan** [comment]
Great

**Comment by ghinda** [comment]
Released `4.0.0`. Text selection is disabled on the `switch-light` "on/off" text now, so swiping the `switch-light` widget will work.

`switch-toggle` still needs the code above, because I didn't want to disble text selection on it.
